hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java

```diff
@@ -193,7 +203,9 @@ public int getPartitionId(Object key) {
 
     @Override
     public int getPartitionCount() {
-        waitForPartitionsFetchedOnce();
+        if (partitionCount == 0) {
```

Comment:
is this if check needed at all?

Reply:
When the check is missing client will do unnecessary steps when switching clusters.
(it will get member list see it is empty and break from the while loop.)

Secondly,  I think the check makes the intention much more clear.